packages/components/shared/src/mixins/observe-attributes-mixin.ts

```diff
@@ -0,0 +1,44 @@
+import { LitElement } from 'lit';
+import { Constructor } from '../types.js';
+
+export interface ObserveAttributesMixinInterface {
+  setTargetElement(target: HTMLElement): void;
+  setObservedAttributes(attributes: string[]): void;
+}
+
+/**
+ * Mixin that is used to rewrite aria attributes in the component (based on the observedAttributes) to the focusable target element.
+ */
+export function ObserveAttributesMixin<T extends Constructor<LitElement>>(
+  constructor: T
+): T & Constructor<ObserveAttributesMixinInterface> {
+  class ObserveAttributesImpl extends constructor {
+    static #observedAttributes: string[] = [];
+
+    #targetElement?: HTMLElement;
+
+    setTargetElement(target: HTMLElement): void {
```

Comment:
when using this in the context of the component that uses the mixin, is this name clear enough? Should "attributes" somehow come back in the name so it's clear it's part of the mixin?